rust_src/src/character.rs

```diff
@@ -0,0 +1,20 @@
+use std::os::raw::c_char;
+use std::ptr;
+
+use lisp;
+
+extern crate libc;
+
+use lisp::{LispObject, LispSubr, EmacsInt};
+
+/// Maximum character code ((1 << CHARACTERBITS) - 1).
+pub const MaxChar: EmacsInt = 0x3FFFFF;
+
```

Comment:
> Why not do the calculation rather than leaving it in a comment?

Will move this from comment to calculation

> We should also confirm that this is the right value on both 32 and 64 bit builds.

Yeah, I will check this